[ET-VK][Ops] dequantization op shaders and impl

ahmtox opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11553
* #11557
* __->__ #11483
* #11369

Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

Commit:
[ET-VK][Ops] dequantization op shaders and impl

Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

manuelcandales commented on backends/vulkan/test/op_tests/dequantize_test.cpp:
```diff
@@ -749,14 +753,40 @@ TEST(
 
 TEST(
     VulkanDequantizePerTensorTest,
-    test_reference_dequantize_per_tensor_int16_to_float) {
```
why did you delete this int16 test?

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantization op shaders and impl"


Creating the dequantize_per_tensor and dequantize_per_token logic shaders and impl which are linked with the testing framework.

Differential Revision: [D76267107](https://our.internmc.facebook.com/intern/diff/D76267107/)

[ghstack-poisoned]
